_posts/2023-03-09-group-equivariant-cnn-1/group-equivariant-cnn-1.Rmd

```diff
@@ -0,0 +1,148 @@
+>
+> 1.  Closure: If $A$ and $B$ are two elements in $G$, then the product $AB$ is also in $G$.
+>
+> 2.  Associativity: The defined multiplication is associative, i.e., for all $A$,$B$,$C$ in $G$, $(AB)C=A(BC)$.
+>
+> 3.  Identity: There is an identity element $I$ (a.k.a. $1$, $E$, or $e$) such that $IA=AI=A$ for every element $A$ in $G$.
+>
+> 4.  Inverse: There must be an inverse (a.k.a. reciprocal) of each element. Therefore, for each element $A$ of $G$, the set contains an element $B=A^{-1}$ such that $AA^{-1}=A^{-1}A=I$.
+
+In action-speak, group elements specify allowable actions; or more precisely, ones that are distinguishable from each other. Two actions can be composed; that's the "binary operation". The requirements now make intuitive sense:
+
+1.  A combination of two actions -- two rotations, say -- is still an action of the same type (a rotation).
+2.  If we have three such actions, it doesn't matter how we group them. (Their order of application has to remain the same, though.)
+3.  One possible action is always the "null action". (Just like in life.) As to "doing nothing", it doesn't make a difference if that happens before or after a "something"; that "something" is always the final result.
+4.  Every action needs to have an "undo button". In the squares example, if I rotate by 180 degrees, and then, by 180 degrees again, I am back in the original state. It is if I had done *nothing*.
+
+Resuming a more "birds-eye view", what we've seen right now is the definition of a group by how its elements act on each other. But if groups are to matter "in the real world", they need to act on something outside (neural network components, for example). How this works is the topic of the following posts, but I'll briefly outline the intuition here.
+
+## Outlook: Group-equivariant CNN
+
+Above, we noted that, in image classification, a *translation*-invariant operation (like convolution) is needed: A $1$ is a $1$ whether moved horizontally, vertically, both ways, or not at all. What about rotations, though? Standing on its head, a digit is still what it is. Conventional convolution does not support this type of action.
+
+We can add to the list of desiderates by specifying a symmetry group. What group? If we wanted to detect squares aligned to the axes, a suitable group would be $C_4$, the cyclic group of order four. (Above, we saw that we needed four elements, and that we could *cycle* through the group.) If, on the other hand, we don't care about alignment, we'd want *any* position to count. In principle, we should end up in the same situation as we did with the sphere. However, images live on discrete grids; there won't be an unlimited number of rotations in practice.
```

Comment:
Again, unless desiderates is used in this specific area of study, I'd find a replacement word.